pkg/k8s/service_cache.go

```diff
@@ -555,7 +555,7 @@ func (s *ServiceCache) mergeServiceUpdateLocked(service *serviceStore.ClusterSer
 
 	// we don't need to check if the current cluster is remote or local,
 	// as externalEndpoints should not have any local cluster endpoints anyway.
-	if service.IncludeExternal && !service.Shared {
+	if !service.Shared {
```

Comment:
do you think we need to check for IncludeExternal as well ? Something like below

```suggestion
	if !service.IncludeExternal || !service.Shared {
```

Reply:
I think it mainly depends on the behavior we want to achieve in case 4) above (whether it should be meaningful at all or not). What I like about the current approach is that you can express the fact that you want to share your endpoints with other clusters marking the service as shared, without using the remote ones in your local cluster (i.e., not marking it as global). Otherwise, I don't think it could be feasible. If we take this route, though, the docs should be probably updated to ack this possibility.

Reply:
Given the behavior agreed upon for 4, I've applied the suggested change. For future reference, the scenario described above (i.e., sharing the endpoints while preferring to use the local ones) could be achieved in the proposed configuration setting the service-affinity to local.